test/extensions/filters/http/ext_authz/ext_authz_integration_test.cc

```diff
@@ -47,6 +49,10 @@ class ExtAuthzGrpcIntegrationTest : public Grpc::GrpcClientIntegrationParamTest,
       setGrpcService(*proto_config_.mutable_grpc_service(), "ext_authz_cluster",
                      fake_upstreams_.back()->localAddress());
 
+      // Override timeout if needed.
+      proto_config_.mutable_grpc_service()->mutable_timeout()->CopyFrom(
```

Comment:
It looks like ext_proc grpc_service: https://github.com/envoyproxy/envoy/blob/5c056957f8742dc1cd056bc5f20a02d28ab480b2/api/envoy/extensions/filters/http/ext_proc/v3/ext_proc.proto#L103 also has this timeout. But it also implemented the filter timeout: https://github.com/envoyproxy/envoy/blob/5c056957f8742dc1cd056bc5f20a02d28ab480b2/api/envoy/extensions/filters/http/ext_proc/v3/ext_proc.proto#L148.

I think one is for the transport layer, and one is for application layer. And it's the application layer timeout which ext_authz is missing.  I guess the ext_proc added the application layer timeout is that even though transport may see the response, but it does not necessarily means the ext_proc filter receives it.

Reply:
Maybe I'm misunderstanding what you're saying. Aren't grpc and http both "application level" protocols?

Would it be possible to write a test that doesn't trigger the transport timeout but would trigger the application timeout? I'm having trouble imagining how this would happen.

Reply:
ext_proc can also configure timeout in [StreamOptions](https://github.com/envoyproxy/envoy/blob/4a34aff94eb0add8b211049d8167f874a2ec233c/source/extensions/filters/http/ext_proc/client_impl.cc#L44-L46).  Also, there are two type of gRPC in Envoy. (a) envoy-gRPC which uses upstream connection management. (b) google gRPC.  It is possible that it is not fully working and there are some gaps(like we are now adding retryPolicy support in StreamOptions). I feel gRPC level timeout is most natural way (if it could work )and we can investigate it for ext_proc later as it is out of scope of this PR.

PS: I think Yanjun mean to say `one is for the transport layer, and one is for application layer` by:  one is gRPC layer (ext_authz approach), one is filter layer where filter uses a envoy timer (ext_proc's approach).

Reply:
I think we've determined that the timeouts already implemented for ext_authz should be sufficient, and all this PR does is add integration tests to demonstrate they are working properly. I'm going to resolve this conversation, but if there is anything else either of you want to bring up feel free to reopen.